modules/mobile/app/models/mobile/v0/adapters/lighthouse_direct_deposit_error.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    module Adapters
+      class LighthouseDirectDepositError
+        def self.parse(response)
+          body = parse_body(response[:body])
+          detail = parse_detail(body)
+          status = parse_status(response[:status], detail)
+
+          errors = [
+            {
+              title: parse_title(body),
+              detail:,
+              code: parse_code(detail),
+              source: data_source
+            }
+          ]
+
+          Lighthouse::DirectDeposit::ErrorResponse.new(status, errors)
+        end
+
+        def self.parse_status(status, detail)
```

Comment:
Is this the only difference between this adapter and the `Lighthouse::DirectDeposit::ErrorResponse`? It seems like a lot of code duplication for just this change. Would it make sense to inherit from that class instead? IMO, it could even be worth changing `Lighthouse::DirectDeposit::ErrorResponse` to have a `status` method. Then this could just be a sub-class that overwrites that one method. That would allow for less code drift.

Reply:
It is the only difference but in the interest of backwards compatibility I thought it would be best to just lock in our own version now and not worry if web wants to change anything about errors. I know Tom wrote the parser because web was wanting errors in a different way so I wouldn't want that to affect us in the future. If you think that isn't a big deal though happy to change

Reply:
Thank you for ignoring the fact that I keep saying `ErrorResponse` when i mean error parser. 

I don't really know what the right thing to do is here. If subclass and they change things, it might impact our code in undesirable ways. If we don't subclass and they change things, we will miss out on changes we may want. So it could be problematic either way. 

My natural inclination is to modify the other team's code to allow overriding because 1) this feels like a lot of duplication 2) we're generally at the mercy of service objects outside of our control so the risk that they may change something we use just seems like business as usual to me. 

That override could be as simple as allowing for an optional `status` arg that overrides the response status. Or it could be to create a separate status method that we can override in our subclass.

Ultimately this is your ticket and I defer to you on it. This can go wrong in either direction and there is no clear right answer.

Reply:
I agree the risk is similar to what we already deal with regularly. Ok just made the changes and now it's subclassed